torch_object_detection_segmenter.py

```diff
@@ -1,31 +1,214 @@
+                target_h = target_w = target_size
+            elif isinstance(target_size, Tuple) and len(target_size) == 2:
+                target_h, target_w = target_size
+            else:
+                raise ValueError(f'target_size should be an integer or a tuple of two integers: {target_size}')
+            w_beg = left
+            h_beg = top
+            if how == 'center':
+                w_beg = int((img_w - target_w) / 2)
+                h_beg = int((img_h - target_h) / 2)
+            elif how == 'random':
+                w_beg = np.random.randint(0, img_w - target_w + 1)
+                h_beg = np.random.randint(0, img_h - target_h + 1)
+            elif how == 'precise':
+                assert (w_beg is not None and h_beg is not None)
+                assert (0 <= w_beg <= (img_w - target_w)), f'left must be within [0, {img_w - target_w}]: {w_beg}'
+                assert (0 <= h_beg <= (img_h - target_h)), f'top must be within [0, {img_h - target_h}]: {h_beg}'
+            else:
+                raise ValueError(f'unknown input how: {how}')
+            if not isinstance(w_beg, int):
+                raise ValueError(f'left must be int number between 0 and {img_w}: {left}')
+            if not isinstance(h_beg, int):
+                raise ValueError(f'top must be int number between 0 and {img_h}: {top}')
+            w_end = w_beg + target_w
+            h_end = h_beg + target_h
+            img = img.crop((w_beg, h_beg, w_end, h_end))
+            return img, h_beg, w_beg
+
+        def _get_input_data(docs: DocumentArray, parameters: dict):
+            traversal_paths = parameters.get('traversal_paths', self._default_traversal_paths)
+            batch_size = parameters.get('batch_size', self._default_batch_size)
+
+            # traverse thought all documents which have to be processed
+            flat_docs = docs.traverse_flat(traversal_paths)
+
+            # filter out documents without images
+            filtered_docs = [doc for doc in flat_docs if doc.blob is not None]
+
+            return _batch_generator(filtered_docs, batch_size)
+
         if not docs:
             return
-        for doc in docs:
-            assert doc.blob.shape[0] == 256
-            doc.embedding = np.random.rand(self._dim)
\ No newline at end of file
+
+
+        batch = _get_input_data(docs, parameters) # a generator of batches of docs
+
+        for i, docs_batch in enumerate(batch):
+            # the blob dimension of imgs/cars.jpg at this point is (2, 681, 1264, 3)
+            # "Ensure the color channel axis is the default axis." i.e. c comes first
+            # e.g. (h,w,c) -> (c,h,w) / (b,h,w,c) -> (b,c,h,w)
+            blob_batch = [_move_channel_axis(d.blob, self.channel_axis,
+                                       self._default_channel_axis) for d in docs_batch]
+
+            all_predictions = self._predict(blob_batch)
+
+            for blob, predictions in zip(blob_batch, all_predictions):
```

Comment:
Add docsbatch to the zip

Reply:
This fixed the issue of duplicate chunks in the docs